cmd/machine-api-operator/client_builder.go

```diff
@@ -24,23 +25,32 @@ func (cb *ClientBuilder) OpenshiftClientOrDie(name string) osclientset.Interface
 	return osclientset.NewForConfigOrDie(rest.AddUserAgent(cb.config, name))
 }
 
+// MachineClientOrDie returns the machine api client interface for machine api objects.
+func (cb *ClientBuilder) MachineClientOrDie(name string) mapiclientset.Interface {
+	return mapiclientset.NewForConfigOrDie(rest.AddUserAgent(cb.config, name))
+}
+
 // NewClientBuilder returns a *ClientBuilder with the given kubeconfig.
 func NewClientBuilder(kubeconfig string) (*ClientBuilder, error) {
+	config, err := getRestConfig(kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ClientBuilder{
+		config: config,
+	}, nil
+}
+
+func getRestConfig(kubeconfig string) (*rest.Config, error) {
 	var config *rest.Config
 	var err error
-
 	if kubeconfig != "" {
 		glog.V(4).Infof("Loading kube client config from path %q", kubeconfig)
 		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
 	} else {
 		glog.V(4).Infof("Using in-cluster kube client config")
 		config, err = rest.InClusterConfig()
 	}
-	if err != nil {
-		return nil, err
-	}
-
-	return &ClientBuilder{
-		config: config,
-	}, nil
+	return config, err
```

Comment:
Not sure if this code needs to handle the `InClusterConfig()` case as it already handled by:

```go
// BuildConfigFromFlags is a helper function that builds configs from a master
// url or a kubeconfig filepath. These are passed in as command line flags for cluster
// components. Warnings should reflect this usage. If neither masterUrl or kubeconfigPath
// are passed in we fallback to inClusterConfig. If inClusterConfig fails, we fallback
// to the default config.
func BuildConfigFromFlags(masterUrl, kubeconfigPath string) (*restclient.Config, error) {
	if kubeconfigPath == "" && masterUrl == "" {
		klog.Warningf("Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.")
		kubeconfig, err := restclient.InClusterConfig()
		if err == nil {
			return kubeconfig, nil
		}
		klog.Warning("error creating inClusterConfig, falling back to default config: ", err)
	}
	return NewNonInteractiveDeferredLoadingClientConfig(
		&ClientConfigLoadingRules{ExplicitPath: kubeconfigPath},
		&ConfigOverrides{ClusterInfo: clientcmdapi.Cluster{Server: masterUrl}}).ClientConfig()
}
```

Reply:
Since this part of logic is not being added in this PR, so i would like to leave it as it is.